src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
```

Comment:
First, if the size of the array is greater than the maximum array size (2^31 - 1), then Arrays.copyOf will throw an
OutOfMemoryError. 
Also, if the array is empty, then the code is attempting to allocate an array with a negative size, which will also result in an OutOfMemoryError
We can avoid error-prone code with a little bit cleaner for example...

        if (i > data.length) {
            Object[] newData = new Object[data.length * 3 / 2 + 1];
            System.arraycopy(data, 0, newData, 0, size);
            data = newData;
        }
        return array = (T[]) new Object[def_capacity];